File: AD-HOC/1542/main.rb
def total_pages(q, d, p)
  (q * d * p) / (p - q)
end

def formatted_print(pages)
  puts "#{pages} #{(pages > 1) ? "paginas" : "pagina"}"
end

quantities = gets.split.map(&:to_i)
while quantities.size > 1
  pages = total_pages(*quantities)
  formatted_print(pages)
  quantities = gets.split.map(&:to_i)
end

File: AD-HOC/1542/main.py
def total_pages(q, d, p):
    return (q * d * p) // (p - q)


def formatted_print(pages):
    print(f'{pages} {"paginas" if pages > 1 else "pagina"}')


quantities = list(map(int, input().split()))
while (len(quantities) > 1):
    pages = total_pages(*quantities)
    formatted_print(pages)
    quantities = list(map(int, input().split()))
